src/cmap/wire_protocol/responses.ts

```diff
@@ -104,6 +104,40 @@ export class MongoDBResponse extends OnDemandDocument {
   // {ok:1}
   static empty = new MongoDBResponse(new Uint8Array([13, 0, 0, 0, 16, 111, 107, 0, 1, 0, 0, 0, 0]));
 
+  /**
+   * Returns true iff:
+   * - ok is 0 and the top-level code === 50
+   * - ok is 1 and the writeErrors array contains a code === 50
+   * - ok is 1 and the writeConcern object contains a code === 50
+   */
+  get isMaxTimeExpiredError() {
+    // {ok: 0, code: 50 ... }
+    const isTopLevel = this.ok === 0 && this.code === MONGODB_ERROR_CODES.MaxTimeMSExpired;
+    if (isTopLevel) return true;
+
+    if (this.ok === 0) return false;
+
+    // {ok: 1, writeConcernError: {code: 50 ... }}
+    const isWriteConcern =
+      this.get('writeConcernError', BSONType.object)?.getNumber('code') ===
+      MONGODB_ERROR_CODES.MaxTimeMSExpired;
+    if (isWriteConcern) return true;
+
+    const writeErrors = this.get('writeErrors', BSONType.array);
+    if (writeErrors?.size()) {
+      for (let i = 0; i < writeErrors.size(); i++) {
```

Comment:
can we rename i to `err` something more descriptive?

Reply:
`i` is the index into the `writeErrors` array whereas `err` might imply I am doing something like a for-of loop where the variable is the value inside the array rather than the index. `i` is pretty typical for indexing, the only other name that would fit would be `index` or `offset` but seems like overkill given the context, no?

Reply:
Other possible names I was thinking were `currErr` or `currWriteErr`? I don't feel too strongly it just was less easy to grok for me with the current naming.

Reply:
Both of those names imply the same thing I was mentioning, that the variable is assigned to the "current error" rather than the index of the error. Is there a name that relates to that detail that you would prefer?

Reply:
Usually I think `currentBlank` implies theres multiple `Blanks` and this is one of them. 

We can stick with `i` if no specific var name gets the point across super clearly - since it is a small loop body anyway.